models/articles.models.js

```diff
@@ -12,46 +12,42 @@ exports.selectArticlesById = (article_id) => {
 };
 
 exports.selectAllArticles = (topic) => {
-  
-  if(topic) {
-    return db.query(`SELECT * FROM articles WHERE topic = $1`, [topic])
-    .then((result) => {
-      if(result.rows.length === 0){
-        return Promise.reject({status: 404, msg: 'article does not exist for matched query'})
-      }
-      return result.rows;
-    })
-  }
-  else {
-    const sqlString = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url,
-    CAST(COUNT(comments.article_id) AS INTEGER ) AS comment_count  
-    FROM articles
-    LEFT JOIN comments ON comments.article_id = articles.article_id
-    GROUP BY articles.article_id
-    ORDER BY created_at DESC;`
-  return db
-    .query(sqlString)
-    .then((result) => {
-      return result.rows;
-    });
+  const queryVals = [];
+  let sqlString = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url,
+  CAST(COUNT(comments.article_id) AS INTEGER ) AS comment_count  
+  FROM articles
+  LEFT JOIN comments ON comments.article_id = articles.article_id`;
+
+  if (topic) {
+    queryVals.push(topic);
+    sqlString += ` WHERE topic = $1`;
   }
-};
 
+  return db.query(sqlString +` GROUP BY articles.article_id
+  ORDER BY created_at DESC;`, queryVals)
+  .then((result) => {
+    if(result.rows.length === 0){
+      return Promise.reject({status: 404, msg: 'topic does not exist'})
+    }
```

Comment:
Great work refactoring this! It's so close! You just need to have a think about how to check if the topic exists here and if it does, it should respond with the empty result.rows